craft_parts/errors.py

```diff
@@ -0,0 +1,58 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2021 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""The exceptions that can be raised when using craft_parts."""
+
+from abc import ABC
+
+
+class CraftPartsError(Exception, ABC):
```

Comment:
Consider modelling it after ProviderError.
https://github.com/canonical/craft-providers/blob/main/craft_providers/errors.py#L74

Which borrows from the `SnapcraftException` pattern of brief/details/resolution.

Reply:
Updated to follow the craft-providers error format (with minor implementation changes).